x-pack/libbeat/common/aws/credentials.go

```diff
@@ -21,15 +22,25 @@ import (
 
 // ConfigAWS is a structure defined for AWS credentials
 type ConfigAWS struct {
-	AccessKeyID          string   `config:"access_key_id"`
-	SecretAccessKey      string   `config:"secret_access_key"`
-	SessionToken         string   `config:"session_token"`
-	ProfileName          string   `config:"credential_profile_name"`
-	SharedCredentialFile string   `config:"shared_credential_file"`
-	Endpoint             string   `config:"endpoint"`
-	RoleArn              string   `config:"role_arn"`
-	AWSPartition         string   `config:"aws_partition"` // Deprecated.
-	ProxyUrl             *url.URL `config:"proxy_url"`
+	AccessKeyID          string              `config:"access_key_id"`
+	SecretAccessKey      string              `config:"secret_access_key"`
+	SessionToken         string              `config:"session_token"`
+	ProfileName          string              `config:"credential_profile_name"`
+	SharedCredentialFile string              `config:"shared_credential_file"`
+	Endpoint             string              `config:"endpoint"`
+	RoleArn              string              `config:"role_arn"`
+	AWSPartition         string              `config:"aws_partition"` // Deprecated.
+	ProxyUrl             *url.URL            `config:"proxy_url"`
+	WebIdentityProvider  WebIdentityProvider `config:",inline"`
+}
+
+type WebIdentityProvider struct {
+	RoleArn       string `config:"aws_role_arn"`
+	TokenFilePath string `config:"aws_web_identity_token_file"`
```

Comment:
I see in my draft [PR](https://github.com/elastic/beats/pull/21087) (I will close it right now), there is also a RoleSession parameter. It's not required but do you think it would be good to add?

Reply:
I will investigate if using the same `RoleSessionName` can cause conflict.
We could also just randomise it and make it unique